silkworm/silkrpc/storage/remote_chain_storage.cpp

```diff
@@ -41,49 +41,34 @@ Task<BlockNum> RemoteChainStorage::highest_block_number() const {
 }
 
 Task<std::optional<BlockNum>> RemoteChainStorage::read_block_number(const Hash& hash) const {
-    silkworm::Block block;
-    const bool success = co_await backend_->get_block({.hash = hash.bytes}, /*.read_senders=*/false, block);
-    std::optional<BlockNum> number;
-    if (success) {
-        number = block.header.number;
-    }
+    const auto number = co_await core::rawdb::read_header_number(reader_, hash);
     co_return number;
 }
 
 Task<bool> RemoteChainStorage::read_block(HashAsSpan hash, BlockNum number, bool read_senders, silkworm::Block& block) const {
-    co_return co_await backend_->get_block({number, hash}, read_senders, block);
+    co_return co_await backend_->get_block(number, hash, read_senders, block);
 }
 
 Task<bool> RemoteChainStorage::read_block(const Hash& hash, BlockNum number, silkworm::Block& block) const {
-    co_return co_await backend_->get_block({number, hash.bytes}, /*.read_senders=*/false, block);
+    co_return co_await backend_->get_block(number, hash.bytes, /*.read_senders=*/false, block);
 }
 
 Task<bool> RemoteChainStorage::read_block(const Hash& hash, silkworm::Block& block) const {
-    co_return co_await backend_->get_block({.hash = hash.bytes}, /*.read_senders=*/false, block);
+    const auto number = co_await core::rawdb::read_header_number(reader_, hash);
+    co_return co_await backend_->get_block(number, hash.bytes, /*.read_senders=*/false, block);
 }
 
-Task<std::optional<BlockHeader>> RemoteChainStorage::read_header(BlockNum number, HashAsArray hash) const {
-    silkworm::Block block;
-    const bool success = co_await backend_->get_block({number, hash}, /*.read_senders=*/false, block);
-    std::optional<BlockHeader> header;
-    if (success) {
-        header = std::move(block.header);
-    }
-    co_return header;
+Task<std::optional<BlockHeader>> RemoteChainStorage::read_header(BlockNum number, HashAsArray& hash) const {
+    Hash hash_type{hash};
+    co_return co_await core::rawdb::read_header(reader_, hash_type, number);
```

Comment:
You need to switch back to the previous implementation to make it work also with remote snapshots